cli/bpmetadata/tfconfig_test.go

```diff
@@ -0,0 +1,227 @@
+package bpmetadata
+
+import (
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"path"
+	"testing"
+)
+
+const (
+	tfTestdataPath = "../testdata/bpmetadata/tf"
+	interfaces     = "interfaces"
+)
+
+func TestTFVariables(t *testing.T) {
+	varTests := []struct {
+		name            string
+		configDirPath   string
+		varName         string
+		wantDescription string
+		wantVarType     string
+		wantDefault     interface{}
+		wantRequired    bool
+	}{
+		{
+			name:            "just name and description",
+			configDirPath:   path.Join(tfTestdataPath, interfaces),
+			varName:         "project_id",
+			wantDescription: "The project ID to host the cluster in",
+			wantRequired:    true,
+		},
+		{
+			name:            "with type and string default",
+			configDirPath:   path.Join(tfTestdataPath, interfaces),
+			varName:         "description",
+			wantDescription: "The description of the cluster",
+			wantVarType:     "string",
+			wantDefault:     "some description",
+			wantRequired:    false,
+		},
+		{
+			name:            "with required as fasle",
+			configDirPath:   path.Join(tfTestdataPath, interfaces),
+			varName:         "regional",
+			wantDescription: "Whether is a regional cluster",
+			wantVarType:     "bool",
+			wantDefault:     true,
+			wantRequired:    false,
+		},
+	}
+
+	for _, tt := range varTests {
+		t.Run(tt.name, func(t *testing.T) {
+			mod, _ := tfconfig.LoadModule(tt.configDirPath)
+			got := getBlueprintVariable(mod.Variables[tt.varName])
```

Comment:
I was worried if it would flake out because the order of the vars/outputs may not exactly be same as the configured test cases but it seems like its all in order. I've changed it to target `getBlueprintInterfaces` instead.

Reply:
Actually it is flaky. 1 time out 10 it fails but not too bad. We think of a better way later as long as this gives us better coverage.

Reply:
@g-awmalik thats a good point. The function should stable sort this otherwise when we render this as markdown this can cause diff between runs.

Reply:
Created b/249674630 to track this